test/flows.spec.js

```diff
@@ -522,4 +523,24 @@ describe('Flows', function() {
       }).type).to.equal('StripeInvalidRequestError');
     });
   });
+
+  describe('FileUpload', function() {
+    it('Allows you to upload a file as a stream', function() {
+      var testFilename = path.join(__dirname, 'resources/data/minimal.pdf');
+      var f = fs.createReadStream(testFilename);
+
+      return expect(
+        stripe.fileUploads.create({
+          purpose: 'dispute_evidence',
+          file: {
+            data: f,
+            name: 'minimal.pdf',
+            type: 'application/octet-stream',
+          },
+        }).then(null, function(error) {
+          return error;
+        })
+      ).to.eventually.have.nested.property('size', 739);
+    });
+  });
 });
```

Comment:
We don't actually test that the error comes back as expected; this should do it:

```
    it('Surfaces stream errors correctly', function(done) {
      var mockedStream = new stream.Readable();
      mockedStream._read = function() {};

      var fakeError = new Error('I am a fake error');

      process.nextTick(function() {
        mockedStream.emit('error', fakeError);
      });

      stripe.fileUploads.create({
        purpose: 'dispute_evidence',
        file: {
          data: mockedStream,
          name: 'minimal.pdf',
          type: 'application/octet-stream',
        },
      }).catch(function(error) {
        expect(error.message).to.equal('An error occurred while attempting to process the file for upload.');
        expect(error.detail).to.equal(fakeError);

        done();
      });
    });
```

We should also test that you can still provide a (non-Stream) file, which is just the same as what you've already got except with `fs.readFileSync()` instead of `fs.createReadStream()`.

Reply:
(I tried doing this in a promisey way but ~I don't trust promises in Mocha~ I couldn't get it to work.  ¯\\\_(ツ)_/¯ :) )

Reply:
added both tests (for synchronous file uploads and testing for the stream error to return properly) - thanks!